### ISSUE TITLE
Compile error when using futures-task v0.3.31

### ISSUE THREAD

**Issue by twosatsmaxi** [zeromq/zmq.rs#199]
```
error[E0310]: the parameter type `S` may not live long enough
   --> /Users/alpesh/.cargo/registry/src/index.crates.io-6f17d22bba15001f/zeromq-0.4.0/src/fair_queue.rs:120:29
    |
120 |             let waker_ref = waker_ref(&waker);
    |                             ^^^^^^^^^^^^^^^^^
    |                             |
    |                             the parameter type `S` must be valid for the static lifetime...
    |                             ...so that the type `S` will meet its required lifetime bounds
    |
help: consider adding an explicit lifetime bound
    |
89  |     S: Stream<Item = T> + Send + 'static,
    |                                +++++++++

```

**Comment by twosatsmaxi** [comment]
with very simple proxy example 
```
    // Create frontend and backend sockets
    let mut in_ = zeromq::RouterSocket::new();
    let mut out_ = zeromq::DealerSocket::new();;

    // Bind the sockets
    in_.bind("tcp://*:5559").await?;
    out_.bind("tcp://*:5560").await?;

    // Create the proxy
    info!("Starting proxy");
    zeromq::proxy(&in_, &out_, None).await?;
    info!("Proxy started");
```

**Comment by PaoloPana** [comment]
Hi @twosatsmaxi, for this issue I created a PR: https://github.com/zeromq/zmq.rs/pull/198

**Comment by jb-alvarado** [comment]
I have the same problem, running `cargo update` breaks zmq.rs. Thanks for your patch @PaoloPana!

**Comment by artob** [comment]
Have the same problem after upgrading to Rust 1.81.

**Comment by davidlattimore** [comment]
The problem isn't the Rust version, it's due to a change in futures-task v0.3.31. @PaoloPana sent a PR, which I've merged. I suspect it might be worthwhile doing a release with the fix, but I'm not sure of the release process for this crate. I could probably just manually `cargo publish`, `git tag` and create release through the github UI, but I'll wait a day or two to see if anyone who has done releases of this in the past jumps in.

**Comment by davidlattimore** [comment]
I've released v0.4.1 here and on crates.io.
